src/lib/wp/index.js

```diff
@@ -133,42 +139,80 @@ function wp(site) {
     }
     return [];
   }
+
+  async function getPostBySlug(slug, lang) {
+    const resources = await getResourcesBySlug("posts", slug, lang, {
+      embed: "true",
+    });
+    const resource = resources[0];
+    if (isEmpty(resource)) {
+      return undefined;
+    }
+
+    // eslint-disable-next-line no-underscore-dangle
+    const embedded = resource._embedded;
+
+    const post = {
+      ...resource,
+      author: embedded.author[0],
+      content: resource.content.rendered,
+      featured_media: embedded["wp:featuredmedia"][0],
+      title: resource.title.rendered,
+      // set thumbail from acf | generated thumbnail | Featured Image
+      thumbnail_image:
+        resource.acf.attributes.thumbnail_image ||
+        resource.featured_image_src ||
+        embedded["wp:featuredmedia"][0] ||
+        null,
+    };
+    return post;
```

Comment:
```suggestion
    const featuredMedia: embedded["wp:featuredmedia"]?.length ? embedded["wp:featuredmedia"][0] : null;
    const post = {
      ...resource,
      author: embedded.author[0],
      content: resource.content.rendered,
      featured_media: featuredMedia,
      title: resource.title.rendered,
      // set thumbnail from acf | generated thumbnail | Featured Image
      thumbnail_image:
        resource.acf.attributes.thumbnail_image ||
        resource.featured_image_src ||
        featuredMedia,
    };
    return post;
```

What's the difference between `resource.featured_image_src` and the normal `wp:featuredmedia`?

Reply:
@kilemensi 
featured_media_src, is a generated thumbnail from the featured media. If the user sets the thumbnail, we use it, if not we use a generated thumbnail from the featured image, if that's not available, we use the featured image.

>I think the methods were arranged as in getPages, getPosts, etc., and looks like you're changing that.

Lint was complaining that the getpostbySlug was used before it was called (I use it to get the posts from pages with posts)
So i sent the method above it.

Reply:
Then move all `getPost` methods to before `getPage` onces... I think they were arranged alphabetically. I don't think any of the `getPost` method call `getPage`.